### PR TITLE
Misc cleanup

### PR DESCRIPTION
**Pull request by sbosley**
Delete some unused junk and clean up our codebase to use the diamond operator and fix a bunch of harmless warnings.

### CODE REVIEW

File: squidb-tests/src/com/yahoo/squidb/sql/InsertTest.java [hunk 1]
```diff
@@ -273,7 +273,7 @@ public void run() {
                 // insert into testModels (firstName, lastName) values ("Jack", "Sparrow"), ("James", "Bond", 007),
                 // ("Bugs", "Bunny");
                 Object[] values1 = new Object[]{"Jack", "Sparrow"};
-                Object[] values2 = new Object[]{"James", "Bond", Integer.valueOf(007)};
```
Comment (jdkoren) [inline squidb-tests/src/com/yahoo/squidb/sql/InsertTest.java]: aw...

Comment (sbosley) [inline squidb-tests/src/com/yahoo/squidb/sql/InsertTest.java]: Haha sorry. I was cleaning up the "unnecessary boxing" warnings and didn't notice the actual context here.

File: squidb/src/com/yahoo/squidb/utility/SquidUtilities.java [hunk 2]
```diff
@@ -186,7 +186,7 @@ private static void addColumnToRowBuilder(StringBuilder builder, String value, i
      * A version of {@link Collection#addAll(Collection)} that works on varargs without calling Arrays.asList, which is
      * a performance and memory boost
      */
-    public static <T> void addAll(Collection<T> collection, T... objects) {
+    public static <T> void addAll(Collection<T> collection, T[] objects) {
```
Comment (jdkoren) [inline squidb/src/com/yahoo/squidb/utility/SquidUtilities.java]: i thought this was intended to be varargs?

Comment (sbosley) [inline squidb/src/com/yahoo/squidb/utility/SquidUtilities.java]: It's really intended as a wrapper to Collections.addAll, but it does a null check. It's never called directly as a varargs method itself, but instead callers pass their own varargs argument or an array, whichever they happen to have.

**Comment by jdkoren** [comment]
LGTM

**Pull request merged**